experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -1,3 +1,189 @@
+@doc raw"""
+    exchange_left!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8) -> AbstractVector{UInt8}
+
+Given a word `w` which is reduced w.r.t. `W`, modify `w` in-place into an equivalent word
+starting with `i` if the exchange condition can be applied, otherwise leave `w` unchanged.
+Finally, return `w`.
+
+!!! warning
+  If `w` is not a reduced decomposition, the behaviour is arbitrary.
+
+See also [`exchange_right!(::WeylGroup, ::AbstractVector{UInt8}, ::UInt8)`](@ref).
+"""
+function exchange_left!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8)
+  if w[1] == i
+    return w
+  end
+
+  root = i
+  for s in 1:length(w)
+    if w[s] == root
+      for n in 2:s
+        w[n] = w[s - 1]
+      end
+      w[1] = i
+      return w
+    end
+    root = W.refl[Int(w[s]), Int(root)]
+  end
+
+  return w
+end
+
+@doc raw"""
+    exchange_right!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8) -> AbstractVector{UInt8}
+
+Given a word `w` which is reduced w.r.t. `W`, modify `w` in-place into an equivalent word
+ending with `i` if the exchange condition can be applied, otherwise leave `w` unchanged.
+Finally, return `w`.
+
+!!! warning
+  If `w` is not a reduced decomposition, the behaviour is arbitrary.
+
+See also [`exchange_left!(::WeylGroup, ::AbstractVector{UInt8}, ::UInt8)`](@ref).
+"""
+# The algorithm is almost the same as explain_rmul.
+# But we don't need a normal form, so this is faster.
+function exchange_right!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8)
```

Comment:
```suggestion
function exchange_right!(W::WeylGroup, w::AbstractVector{UInt8}, i::UInt8)
  # The algorithm is almost the same as explain_rmul
  # But we don't need a normal form, so this is faster
```
putting the comment between docstring and signature disconnects the two, i.e. the docstring is not attached to the method here. Therefore, put the comment inside the function